api/base/serializers.py

```diff
@@ -55,24 +56,34 @@ def _url_val(val, obj, serializer, **kwargs):
 
 
 class IDField(ser.CharField):
+    """
+    ID field that validates that 'id' in the request body is the same as the instance 'id' for single requests
+    """
     def __init__(self, **kwargs):
         kwargs['label'] = 'ID'
         super(IDField, self).__init__(**kwargs)
 
     def to_internal_value(self, data):
         update_methods = ['PUT', 'PATCH']
-        if self.context['request'].method in update_methods:
+        request = self.context['request']
+        if request.method in update_methods and not isinstance(request.data, list):
```

Comment:
Can you explain what the old check is doing? Wondering if there needs to be an equivalent check for list items?